tests/e2e/actions.go

```diff
@@ -268,72 +391,225 @@ type SubmitConsumerAdditionProposalAction struct {
+	err = json.Unmarshal(bz, txResponse)
+	if err != nil {
+		log.Fatalf("unmarshalling tx response on update-consumer: %s, json: %s", err.Error(), string(bz))
+	}
+
+	if txResponse.Code != 0 {
+		log.Fatalf("sending update-consumer transaction failed with error code %d, Log:'%s'", txResponse.Code, txResponse.RawLog)
+	}
+	tr.waitBlocks(ChainID("provi"), 2, 10*time.Second)
+}
+
+// CreateConsumer creates a consumer chain and returns its consumer-id
+func (tr Chain) CreateConsumer(providerChain, consumerChain ChainID, validator ValidatorID, metadata types.ConsumerMetadata, initParams *types.ConsumerInitializationParameters, powershaping *types.PowerShapingParameters) string {
+
+	chainID := string(tr.testConfig.chainConfigs[consumerChain].ChainId)
+	rec := types.MsgCreateConsumer{
+		ChainId:                  chainID,
+		Metadata:                 metadata,
+		InitializationParameters: initParams,
+		PowerShapingParameters:   powershaping,
+	}
+
+	content, err := json.Marshal(rec)
+	if err != nil {
+		log.Fatalf("failed marshalling ConsumerRegistrationRecord: %s", err.Error())
+	}
+	jsonFile := "/create_consumer.json"
+	bz, err := tr.target.ExecCommand(
+		"/bin/bash", "-c", fmt.Sprintf(`echo '%s' > %s`, content, jsonFile),
+	).CombinedOutput()
+	if err != nil {
+		log.Fatal(err, "\n", string(bz))
+	}
+
+	// Send consumer chain registration
+	cmd := tr.target.ExecCommand(
+		tr.testConfig.chainConfigs[providerChain].BinaryName,
+		"tx", "provider", "create-consumer", jsonFile,
+		`--from`, `validator`+fmt.Sprint(validator),
+		`--chain-id`, string(tr.testConfig.chainConfigs[providerChain].ChainId),
+		`--home`, tr.getValidatorHome(providerChain, validator),
+		`--gas`, `900000`,
+		`--node`, tr.getValidatorNode(providerChain, validator),
+		`--keyring-backend`, `test`,
+		"--output", "json",
+		`-y`,
+	)
+
+	fmt.Println("Create consumer", "cmd", cmd.String())
+	fmt.Println("Create consumer", "json", string(content))
+
+	bz, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatal("create consumer failed ", "error: ", err, "output: ", string(bz))
+	}
+
+	fmt.Println("create consumer output=", string(bz))
+
+	txResponse := &TxResponse{}
+	err = json.Unmarshal(bz, txResponse)
+	if err != nil {
+		log.Fatalf("unmarshalling tx response on consumer-registration: %s, json: %s", err.Error(), string(bz))
+	}
+
+	if txResponse.Code != 0 {
+		log.Fatalf("sending transaction failed with error code %d, Log:'%s'", txResponse.Code, txResponse.RawLog)
+	}
+
+	// TODO: introduce waitForTx
+	tr.waitBlocks(providerChain, 2, 10*time.Second)
+
+	// Get Consumer ID from transaction
+	cmd = tr.target.ExecCommand(
+		tr.testConfig.chainConfigs[providerChain].BinaryName,
+		"query", "tx", txResponse.TxHash,
+		`--node`, tr.getValidatorNode(providerChain, validator),
+		"--output", "json",
+	)
+	bz, err = cmd.CombinedOutput()
+	if err != nil {
+		log.Fatal("not able to query tx containing consumer registration: cmd:", cmd, "err:", err.Error(), "out:", string(bz))
```

Comment:
It's not a registration anymore. I guess, due to the ongoing changes on the permissionless codebases, old terms like this might appear around. Can we search for them and replace them accordingly?